mapactionpy_arcmap/MapChef.py

```diff
@@ -56,25 +58,36 @@ def cook(self, productName, countryName):
+            if (properties is not None):
                 layerFilePath = os.path.join(self.layerDirectory, (properties.layerName + ".lyr"))
                 if (os.path.exists(layerFilePath)):
                     self.dataFrame = arcpy.mapping.ListDataFrames(self.mxd, properties.mapFrame)[0]
                     layerToAdd = arcpy.mapping.Layer(layerFilePath)
-                    dataFilePath= os.path.join(self.root, "GIS", "2_Active_Data", properties.sourceFolder)
+                    # APS 06/09/2019 Please remove hardcoded path. See github comment
+                    dataFilePath = os.path.join(self.root, "GIS", "2_Active_Data", properties.sourceFolder)
                     if (os.path.isdir(dataFilePath)):
+                        # APS 06/09/2019 I do not understand what you are testing here
+                        # (eg ` if ("/" not in properties.regExp):`).
+                        # Why is a forward slash required in a regex for a filename/featureclass name?
                         if ("/" not in properties.regExp):
                             onlyfiles = [f for f in listdir(dataFilePath) if isfile(join(dataFilePath, f))]
                             for fileName in onlyfiles:
                                 if re.match(properties.regExp, fileName):
                                     self.dataFrame = arcpy.mapping.ListDataFrames(self.mxd, properties.mapFrame)[0]
                                     dataFile = os.path.join(dataFilePath, fileName)
-                                    mapResult.added = self.addToLayer(self.dataFrame, dataFile, layerToAdd, properties.definitionQuery, properties.display, properties.labelClasses, countryName)
+                                    mapResult.added = self.addToLayer(
+                                        self.dataFrame, dataFile, layerToAdd, properties.definitionQuery,
+                                        properties.display, properties.labelClasses, countryName)
                                     mapResult.dataSource = dataFile
-                                    if (mapResult.added == True):
+                                    if mapResult.added:
                                         mapResult.message = "Layer added successfully"
                                     else:
-                                        mapResult.message = "Unexpected schema.  Could not evaluate expression: " + properties.definitionQuery
+                                        # APS 06/09/2019 Are you sure that this error can only be reached due to a 
+                                        # schema error? Would an error message along the lines of 
+                                        # "Error adding {layerName}. Possibly due to schema error or other cause" 
+                                        # be appropriate? See github comment
+                                        mapResult.message = ("Unexpected schema.  Could not evaluate expression: "
```

Comment:
Are you sure that this error can only be reached due to a schema error? Would an error message along the lines of "Error adding {layerName}. Possibly due to schema error or other cause" be appropriate. Alternatively something like:
```
if mapResult.added:
	mapResult.message = "Layer added successfully"
elif test_for_schema_error(layerToAdd):
	mapResult.message = ("Unexpected schema.  Could not evaluate expression: "
		             + properties.definitionQuery)
else:
	mapResult.message = "Unknown error adding layer : " + layerToAdd
```